app/services/apply_api/import_application.rb

```diff
@@ -31,11 +34,21 @@ def application
     end
 
     def provider_code
-      @provider_code ||= application_data["attributes"]["course"]["training_provider_code"]
+      @provider_code ||= course_attributes("training_provider_code")
     end
 
     def apply_id
       @apply_id ||= application_data["id"]
     end
+
+    def provider_a_hei?
+      course_attributes("training_provider_type") == "university"
+    end
+
+    def course_attributes(attribute_name)
+      application_data["attributes"]["course"][attribute_name]
+    rescue NoMethodError
+      raise ApplyApiMissingDataError
```

Comment:
We could add the message here. Something like... 
```suggestion
      raise ApplyApiMissingDataError.new("Apply application_id #{apply_id} could not be imported")
```

Reply:
👍